ovos_core/intent_services/commonqa_service.py

```diff
@@ -1,51 +1,61 @@
 import re
-from threading import Lock, Event
-
 import time
+import ovos_core.intent_services
+from dataclasses import dataclass
 from itertools import chain
+from threading import Event
+from typing import Dict
+from ovos_bus_client.session import SessionManager
 from ovos_bus_client.message import Message, dig_for_message
-
-import ovos_core.intent_services
 from ovos_utils import flatten_list
 from ovos_bus_client.apis.enclosure import EnclosureAPI
-from ovos_utils.log import LOG
+from ovos_utils.log import LOG, log_deprecation
 from ovos_bus_client.util import get_message_lang
 from ovos_workshop.resource_files import CoreResources
+from ovos_config.config import Configuration
+
 
-EXTENSION_TIME = 10
+@dataclass
+class Query:
+    session_id: str
+    query: str
+    replies: list = None
+    extensions: list = None
+    query_time: float = time.time()
+    timeout_time: float = time.time() + 1
+    responses_gathered: Event = Event()
+    completed: Event = Event()
+    answered: bool = False
 
 
 class CommonQAService:
-    """Intent Service handling common query skills.
-    All common query skills answer and the best answer is selected
-    This is in contrast to triggering best intent directly.
-    """
+    # Left for `mycroft` backwards-compat.
+    _EXTENSION_TIME = 10
```

Comment:
can you explain this one? how is this backwards compat with mycroft if mycroft had this in a skill?